cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -441,13 +449,18 @@ func (ks *kube2sky) updateService(oldObj, newObj interface{}) {
 	ks.newService(newObj)
 }
 
-func newEtcdClient(etcdServer string) (*etcd.Client, error) {
+func newEtcdClient(config *etcdConfig) (*etcd.Client, error) {
 	var (
 		client *etcd.Client
 		err    error
 	)
+	etcdServer := config.ServerList[0]
+	httpClient, err := config.newHTTPClient()
+	if err != nil {
+		return nil, fmt.Errorf("unable to create HTTP client: %v", err)
+	}
 	for attempt := 1; attempt <= maxConnectAttempts; attempt++ {
```

Comment:
I mostly want to avoid building an http client two ways. Any connectivity check that is done should be done using the client we'll actually be using for etcd calls